src/Phpunit/DatabaseTestListener.php

```diff
@@ -1,225 +0,0 @@
-<?php
-
-/*
- * This file is part of the Symfony CMF package.
- *
- * (c) 2011-2017 Symfony CMF
- *
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
-namespace Symfony\Cmf\Component\Testing\Phpunit;
-
-use Doctrine\Common\DataFixtures\Purger;
-use PHPUnit\Framework\AssertionFailedError;
-use PHPUnit\Framework\Test;
-use PHPUnit\Framework\TestListener;
-use PHPUnit\Framework\TestSuite;
-use PHPUnit\Framework\Warning;
-use Symfony\Component\Process\PhpExecutableFinder;
-use Symfony\Component\Process\Process;
-use Symfony\Component\Process\ProcessUtils;
-
-class DatabaseTestListener implements TestListener
-{
-    protected static $currentSuite;
-
-    private $processCallable;
-
-    private $prefix = [];
-
-    public function __construct(callable $processCallable = null)
-    {
-        $phpExecutableFinder = new PhpExecutableFinder();
-        $phpExecutable = $phpExecutableFinder->find(false);
-        if (false === $phpExecutable) {
-            throw new \RuntimeException('No PHP executable found on the current system.');
-        }
-
-        // Symfony 2.3 does not support array prefix, so we have to implement it ourselves
-        $this->prefix = [$phpExecutable, __DIR__.'/../../bin/console'];
-        $this->processCallable = $processCallable;
-    }
-
-    public function getProcess($arguments)
-    {
-        if (is_callable($this->processCallable)) {
-            $callable = $this->processCallable;
-
-            return $callable($arguments);
-        }
-
-        if (is_array($arguments) && method_exists(ProcessUtils::class, 'escapeArgument')) {
-            $arguments = implode(' ', array_map([ProcessUtils::class, 'escapeArgument'], $arguments));
-        }
-
-        return new Process($arguments);
-    }
-
-    public function addError(Test $test, \Throwable $e, float $time): void
-    {
-    }
-
-    public function addFailure(Test $test, AssertionFailedError $e, float  $time): void
-    {
-    }
-
-    public function addWarning(Test $test, Warning $e, float $time): void
-    {
-    }
-
-    public function addIncompleteTest(Test $test, \Throwable $e, float $time): void
-    {
-    }
-
-    public function addSkippedTest(Test $test, \Throwable $e, float $time): void
-    {
-    }
-
-    public function addRiskyTest(Test $test, \Throwable $e, float $time): void
-    {
-    }
-
-    public function startTest(Test $test): void
-    {
-        switch (static::$currentSuite->getName()) {
-            case 'orm':
-                $db = $test->getDbManager('ORM');
-                $purger = new Purger\ORMPurger($db->getOm());
-
-                break;
-
-            case 'phpcr':
-                $db = $test->getDbManager('PHPCR');
-                $purger = new Purger\PHPCRPurger($db->getOm());
```

Comment:
should we not keep purging the database between tests? i agree that the setup is better done with the makefile, but things done between each test can not be moved to the makefile. we could provide a base test or trait of course, but not sure if that is better.

Reply:
We could keep that possibibility, to use it, but none of our packages currently uses it.

Reply:
I would have no problem to keep it, i just would have to make some minor tweaks to stay compatible with the interface